accessibility_monitoring_platform/apps/common/templates/common/metrics/case.html

```diff
@@ -0,0 +1,131 @@
+                    <svg
+                        aria-hidden="true"
+                        xmlns="http://www.w3.org/2000/svg"
+                        viewBox="0 0 {{ metric.chart_width }} {{ metric.chart_height }}"
+                        width="100%"
+                        height="{{ metric.chart_height }}"
+                        stroke-width="3"
+                        fill="none"
+                    >
+                        <g stroke="#1d70b8" transform="translate(50)">
+                            <g stroke="currentColor">
+                                <line x1="0" x2="{{ metric.last_x_position }}" y1="{{ metric.graph_height }}" y2="{{ metric.graph_height }}"></line>
+                                {% for row in metric.all_table_rows %}
+                                    <line x1="{{ row.x }}" x2="{{ row.x }}" y1="{{ metric.graph_height }}" y2="{{ metric.x_axis_tick_y2 }}"></line>
+                                {% endfor %}
+                                <line x1="0" x2="0" y1="0" y2="{{ metric.graph_height }}"></line>
+                                {% for y_axis_label in metric.y_axis_labels %}
+                                    <line x1="-10" x2="0" y1="{{ y_axis_label.y }}" y2="{{ y_axis_label.y }}"></line>
+                                {% endfor %}
+                                <g stroke-width="1" fill="currentColor">
+                                    {% for row in metric.all_table_rows %}
+                                        <text text-anchor="middle" x="{{ row.x }}" y="{{ metric.x_axis_label_1_y }}">{{ row.month_date|date:"M" }}</text>
+                                        {% if row.month_date.month == 1 %}
+                                            <text text-anchor="middle" x="{{ row.x }}" y="{{ metric.x_axis_label_2_y }}">{{ row.month_date.year }}</text>
+                                        {% endif %}
+                                    {% endfor %}
+                                    {% for y_axis_label in metric.y_axis_labels %}
+                                        <text alignment-baseline="middle" text-anchor="end" x="-12" y="{{ y_axis_label.y }}">{{ y_axis_label.label }}</text>
+                                    {% endfor %}
+                                </g>
+                            </g>
+                            <polyline
+                                points="
+                                {% for row in metric.previous_month_rows %}
+                                    {{ row.x }},{{ row.y }}
+                                {% endfor %}
+                                "/>
+                            <polyline
+                                stroke-dasharray="5"
+                                points="
+                                {% for row in metric.current_month_rows %}
+                                    {{ row.x }},{{ row.y }}
+                                {% endfor %}
+                                "/>
+                        </g>
+
```

Comment:
This is really cool. I'm just concerned it might be difficult to maintain long-term. Can a CSS/JS library do this work for us?

Reply:
Did you have a specific library in mind? Is there one which GDS has approved for this use case?

Drawing `line`s and `polyline`s in SVG if fairly trivial. I'd like to try to implement the full set of charts in this fashion. If the requirements make that too difficult then we use a library instead. This implementation is still an early draft, I can make it more maintainable once I have the full set of requirements.

The information needed by any library to render the charts is that same as that used by the SVG, just in a different arbitrary format.